crates/experimentation_platform/src/api/experiments/types.rs

```diff
@@ -129,6 +135,68 @@ pub enum ContextBulkResponse {
     MOVE(ContextPutResp),
 }
 
+/********** Applicable Variants API Type *************/
+#[derive(Debug)]
+pub struct ApplicableVariantsQuery {
+    pub context: Map<String, Value>,
+    pub toss: i8,
+}
+
+impl TryFrom<Map<String, Value>> for ApplicableVariantsQuery {
+    type Error = String;
+    fn try_from(mut value: Map<String, Value>) -> Result<Self, Self::Error> {
+        let toss = value
+            .remove("toss")
+            .and_then(|toss| toss.as_i64())
+            .and_then(|toss| {
+                if -1 <= toss && toss <= 100 {
+                    Some(toss as i8)
+                } else {
+                    None
+                }
+            })
+            .ok_or_else(|| {
```

Comment:
shouldn't toss be optional and default should be -1 ?

Reply:
get_applicable_variant is based on toss itself, toss is mandatory here

Reply:
> shouldn't toss be optional and default should be -1 ?

get_applicable_variant is based on toss itself, toss is mandatory here